### DISCUSSION TITLE
Too many outposts are not converted in radiation

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#14149] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Even when all map is covered with radiation 15 of 65 outpost are still alive, why? They are not even last ouposts in biome 
Is this a bug or intended? 

I would expect only 3 (MinimumOutpostAmount) last outposts on the map to be alive
![20240617154239_1](https://github.com/FakeFishGames/Barotrauma/assets/122838333/e8c3848a-0c64-432a-9c7e-aa368503cbb9)

### Reproduction steps

Start campaign with radiation
Togglecampaignteleport
Jump around the map until it all covered

If you have lua you can speed up the process with these commands
```
cl_lua "LuaUserData.RegisterType('Barotrauma.RadiationParams')"
cl_lua "Game.GameSession.Map.Radiation.Params.RadiationStep = 300"
cl_lua "Game.GameSession.Map.Radiation.Params.CriticalRadiationThreshold = 0;"
```

I could also make xml mod with faster radiation params if you want

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
I think i'm starting to see the pattern:
All alive outposts are either left gate outposts or habitation outposts

Perhaps newly created "habitation outposts" not added to some location list properly

Gate outposts imho also shouldn't be alive because they open their gates when irradiated, so their aliveness is no longer needed

**Comment by Regalis11** [comment]
Are you sure they are actually alive? It seems getting irradiated doesn't actually change the location type nor the icon on the map, but if you go and visit the outpost, it seems to be abandoned.

**Reply by SomeRandomNoobKekeke** [reply]
I'll check...

**Reply by SomeRandomNoobKekeke** [reply]
Hmm...
They are in fact abandoned, but displayed as alive on the map even after a few rounds
Mb it depends on RadiationStep, i used RadiationStep=3000 for this test and look how many of them there
On default settings it's 1-2 per biome
And again, all of them simple habitation or gate outposts
![20240618222339_1](https://github.com/FakeFishGames/Barotrauma/assets/122838333/e2e6f7d4-e4ba-4445-bf93-0e8ac2d7be0d)
![20240618222201_1](https://github.com/FakeFishGames/Barotrauma/assets/122838333/8aab6c8b-e0c2-4fa2-b04a-317de93d8e28)

**Reply by SomeRandomNoobKekeke** [reply]
I used RadiationStep=10000 and took screenshots before and after 2 steps, all habitation outposts preserved :BaroDev:
https://github.com/FakeFishGames/Barotrauma/assets/122838333/a9319e82-1df9-47b8-a04a-dedaf56640ce
https://github.com/FakeFishGames/Barotrauma/assets/122838333/a8ab7986-2872-4b4b-a3d8-0ed4db7d2b2c

**Comment by Regalis11** [comment]
Doesn't seem to be a bug, radiation doesn't actually change the location type nor the icon on the map, just makes it generate as an abandoned outpost.

**Reply by SomeRandomNoobKekeke** [reply]
> radiation doesn't actually change the icon on the map

It does change icon for all outposts except for gate and 1-2 random habitation outposts for unknown reasons

**Comment by SomeRandomNoobKekeke** [comment]
@Regalis11 Ok, i figured it out, there's two problems here:
### 1. LocationType "Outpost" has `replaceinradiation="Abandoned"` but missing `<RadiationSymbol texture="Content/Map/MapAtlas.png" sourcerect="896,512,128,128" color="245,105,105" />`

If you add it, they will appear as abandoned as they should
Although Idk why only a fraction of them breaks, i don't want to dive into this rabbit hole of outpost transformations :AwareDev:

### 2. Gate outposts are not transformed in radiation 
They don't just appear alive, they are in fact alive, https://github.com/FakeFishGames/Barotrauma/discussions/14149#discussioncomment-9810054 was disinfo 
This is happening because you unlock connections for `location.IsGateBetweenBiomes`, but skip `location.TurnsInRadiation++;`
I think removing `continue;` here should fix it
https://github.com/FakeFishGames/Barotrauma/blob/230d1b6e7887cd354bdf0bacc3952c2d3068c6b0/Barotrauma/BarotraumaShared/SharedSource/Map/Map/Radiation.cs#L65-L90

**Reply by SomeRandomNoobKekeke** [reply]
And ofc you didn't ask for it but here's [xml mod to test it](https://steamcommunity.com/sharedfiles/filedetails/?id=3270852174)

**Comment by SomeRandomNoobKekeke** [comment]
@SomeRandomNoobKekeke 
Thank you for the report! Fixed in https://steamcommunity.com/sharedfiles/filedetails/?id=3269363928 (update in my workshop mod).
